File: Implementation/python/10_단어뒤집기.py
# https://www.acmicpc.net/problem/9093
# 1

for _ in range(int(input())):
    S = input()
    result, temp = "", ""

    for s in S:
        if s == ' ':
            result += temp[::-1] + ' '
            temp = ""
        else:
            temp += s
    print(result + temp[::-1])

File: Implementation/c++/10_단어뒤집기.cpp
// https://www.acmicpc.net/problem/9093
// 1

#include <iostream>
#include <algorithm>
#include <sstream>
#include <vector>

using namespace std;

int main()
{
	int test_case;
	string s;
	cin >> test_case;
	getline(cin, s);

	for (int t = 0; t < test_case; t++) {
		getline(cin, s);
		istringstream iss(s);
		vector<string> vs;
		while (getline(iss, s, ' ')) {
			reverse(s.begin(), s.end());
			vs.push_back(s);
		}

		for (int i = 0; i < vs.size(); i++)
			cout << vs[i] << ' ';
		cout << '\n';
	}
}